qiskit/providers/fake_provider/fake_generic.py

```diff
@@ -0,0 +1,535 @@
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Generic fake BackendV2 class"""
+
+from __future__ import annotations
+import warnings
+
+from collections.abc import Iterable
+import numpy as np
+
+from qiskit import pulse
+from qiskit.pulse.instruction_schedule_map import InstructionScheduleMap
+from qiskit.circuit import QuantumCircuit, Instruction
+from qiskit.circuit.controlflow import (
+    IfElseOp,
+    WhileLoopOp,
+    ForLoopOp,
+    SwitchCaseOp,
+    BreakLoopOp,
+    ContinueLoopOp,
+)
+from qiskit.circuit.library.standard_gates import get_standard_gate_name_mapping
+from qiskit.exceptions import QiskitError
+from qiskit.transpiler import CouplingMap, Target, InstructionProperties, QubitProperties
+from qiskit.providers import Options
+from qiskit.providers.basicaer import BasicAer
+from qiskit.providers.backend import BackendV2
+from qiskit.providers.models import (
+    PulseDefaults,
+    Command,
+)
+from qiskit.qobj import PulseQobjInstruction, PulseLibraryItem
+from qiskit.utils import optionals as _optionals
+
+# Noise default values/ranges for duration and error of supported
+# instructions. There are two possible formats:
+# - (min_duration, max_duration, min_error, max_error),
+#   if the defaults are ranges.
+# - (duration, error), if the defaults are fixed values.
+_NOISE_DEFAULTS = {
+    "cx": (1e-8, 9e-7, 1e-5, 5e-3),
+    "ecr": (1e-8, 9e-7, 1e-5, 5e-3),
+    "cz": (1e-8, 9e-7, 1e-5, 5e-3),
+    "id": (3e-8, 4e-8, 9e-5, 1e-4),
+    "rz": (0.0, 0.0),
+    "sx": (1e-8, 9e-7, 1e-5, 5e-3),
+    "x": (1e-8, 9e-7, 1e-5, 5e-3),
+    "measure": (1e-8, 9e-7, 1e-5, 5e-3),
+    "delay": (None, None),
+    "reset": (None, None),
+}
+
+# Ranges to sample qubit properties from.
+_QUBIT_PROPERTIES = {
+    "dt": 0.222e-9,
+    "t1": (100e-6, 200e-6),
+    "t2": (100e-6, 200e-6),
+    "frequency": (5e9, 5.5e9),
+}
+
+# The number of samples determines the pulse durations of the corresponding
+# instructions. This default defines pulses with durations in multiples of
+# 16 for consistency with the pulse granularity of real IBM devices, but
+# keeps the number smaller than what would be realistic for
+# manageability. If needed, more realistic durations could be added in the
+# future (order of 160dt for 1q gates, 1760dt for 2q gates and measure).
+_PULSE_LIBRARY = [
+    PulseLibraryItem(name="pulse_1", samples=np.linspace(0, 1.0, 16, dtype=np.complex128)),  # 16dt
+    PulseLibraryItem(name="pulse_2", samples=np.linspace(0, 1.0, 32, dtype=np.complex128)),  # 32dt
+    PulseLibraryItem(name="pulse_3", samples=np.linspace(0, 1.0, 64, dtype=np.complex128)),  # 64dt
+]
+
+
+class GenericFakeBackend(BackendV2):
```

Comment:
I do wonder if there is a better name we can use here as this class isn't super generic, like the `BackendV2` abstract interface is more generic backend constructor than the interface this class gives you.  Based on the class docstring I'd be inclined to suggest `ConfigurableFakeBackend` but that already existed and I'm sure there is a potential conflict there.

Reply:
Personally I see the purpose of this as being the minimal implementer of `BackendV2`, and if the class is going to be named, then `GenericBackendV2` is an accurate description of it.  I think the class _constructor_ is configurable in a composition sense, but the produced object is not.

Reply:
I would also prefer `Configurable`, also ruled out because of the conflicts with the one that already existed (tbf, now it's been removed, so it's a bit less impossible). There's always the option to call it `ConfigurableFakeBackendV2`, I normally prefer not to add the V2 if the V1 doesn't exist, but in this case this is a workaround to a different problem... not sure.

Reply:
> Personally I see the purpose of this as being the minimal implementer of `BackendV2`

@jakelishman this class bakes-in running on Aer / BasicAer though, which isn't something that real implementations of `BackendV2` would do, correct?

Reply:
It's a fair comment. Maybe it's not the absolute most minimal implementer, but given how free-form `Result` is, the most minimal isn't super usable for anything.

To be a faithful implementor, it has to bake in _some_ way of running - the data made available to `transpile` isn't the only part of it. As long as we don't actually expose the simulator publicly (I commented on that elsewhere, though I imagine I could be convinced to leave it), I don't think it's violating the concept of a minimal implementer for the `run` to be one particular implementation, and whether or not the class is named, we'd have to document if the output can be relied upon to be Aer or not as part of the interface.

I think defaulting to Aer/BasicAer is generally the right level of trade-off, especially given what people used this for before. The class _could_ be more general, but we the again, we can keep making that argument all the way down to "define your own subclass".